### ISSUE TITLE
Menu button in the chat history area does not indicate its state

### ISSUE THREAD

**Issue by derekjackson-das** [danny-avila/LibreChat#3823]
**Issue Description**: The menu button in the chat history area of the LibreChat interface does not provide a programmatic indication of whether it is expanded or collapsed. Additionally, it does not specify that it has a menu popup or provide a reference to that control. This lack of state information affects accessibility, particularly for users relying on screen readers.

Related to #3821 

**WCAG Criteria**: [4.1.2 Name, Role, Value - A](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)

Screenshot or Screen Recording: 
[Screen Reader Review](https://share.zight.com/eDuZolvL
)

<img width="1387" alt="Screenshot 2024-08-28 at 12 24 45 PM" src="https://github.com/user-attachments/assets/4a383e5c-fc05-4a2c-ada4-3b4ca8936ee2">

**Snippet**:

`<button class="rounded-md inline-flex items-center justify-center text-sm font-medium focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-transparent underline-offset-4 hover:underline text-gray-600 dark:text-gray-400 hover:bg-transparent dark:hover:bg-transparent z-10 h-7 w-7 border-none p-0 transition-all duration-200 ease-in-out" id="conversation-menu-button" aria-label="conversation-menu-button">`

Notes:Add the aria-haspopup="menu", aria-expanded="true/false", and aria-controls="[menuitemsContainer#id]" attributes. These attributes will identify that the button controls a popup menu, indicate whether the menu is open or closed, and reference the associated menu element.

For guidance on how a menu button should function, refer to the [ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/). While the implementation details may vary, the conveyed information and user interface should align with these practices.

Also see the menu button for Account Settings in the sidebar. It has these features implemented. 

<img width="1682" alt="Screenshot 2024-08-28 at 2 16 27 PM" src="https://github.com/user-attachments/assets/b52fabd7-d12d-4214-b48f-da38474bc619">
